score.go

```diff
@@ -153,12 +149,56 @@ func (d *indexData) scoreFileUsingBM25(fileMatch *FileMatch, doc uint32, cands [
 	// Use standard parameter defaults (used in Lucene and academic papers)
 	k, b := 1.2, 0.75
 	sumTf := 0.0 // Just for debugging
-	score := 0.0
-	for _, freq := range termFreqs {
+
+	tfs := make(termFrequencyScore)
+
+	for term, freq := range termFreqs {
 		tf := float64(freq)
 		sumTf += tf
-		score += ((k + 1.0) * tf) / (k*(1.0-b+b*L) + tf)
+
+		// Invariant: the keys of df are the union of the keys of tfs over all files.
+		df[term] += 1
```

Comment:
Double-checking: we don't have any setting that allows within-shard parallelism right? So there could never be contention on this `df` map.

Reply:
Correct, we don't have within-shard parallelism.